astroid/brain/brain_hypothesis.py

```diff
@@ -36,11 +38,14 @@ def is_decorated_with_st_composite(node) -> bool:
     return False
 
 
-def remove_draw_parameter_from_composite_strategy(node):
+def remove_draw_parameter_from_composite_strategy(node: FunctionDef) -> FunctionDef:
     """Given that the FunctionDef is decorated with @st.composite, remove the
     first argument (`draw`) - it's always supplied by Hypothesis so we don't
     need to emit the no-value-for-parameter lint.
     """
+    if TYPE_CHECKING:
```

Comment:
I think it is better to actually do this assert always. `mypy` is showing us we have unsafe code, we should make it safe by asserting our assumption so we can see where they don't hold and handle it accordingly.